src/kOS/Utilities/Utils.cs

```diff
@@ -63,90 +63,6 @@ public static bool IsValidRotation(Quaternion quaternion)
-
-        public static double ProspectForResource(string resourceName, Part part, IEnumerable<FuelLine> lines, int rDepth, ref List<Part> visited)
-        {
-#pragma warning disable 162
-            const bool DEBUG_WALK = false; // set to true to enable the logging of the recursive walk.
-            var indent = new String(',', rDepth);
-
-            if (DEBUG_WALK) Debug.Log(indent + "ProspectForResource( " + resourceName + ", " + part.uid() + ":" + part.name + ", ...)");
-            double ret = 0;
-
-            if (visited.Contains(part))
-            {
-                if (DEBUG_WALK) Debug.Log(indent + "- Already visited, truncate recurse branch here.");
-                return 0;
-            }
-
-            visited.Add(part);
-
-            foreach (PartResource resource in part.Resources)
-            {
-                if (String.Equals(resource.resourceName, resourceName, StringComparison.CurrentCultureIgnoreCase))
-                {
-                    ret += resource.amount;
-                }
-            }
-
-            foreach (var attachNode in GetActualAttachedNodes(part))
-            {
-                if (DEBUG_WALK) Debug.Log(indent + "- AttachNode " + attachNode.id);
-                if (!attachNode.ResourceXFeed) continue;
-
-                if (DEBUG_WALK) Debug.Log(indent + "- - It is an xfeed-able attachnode.");
-                if (attachNode.attachedPart == null || (!attachNode.attachedPart.fuelCrossFeed)) continue;
-
-                if (DEBUG_WALK) Debug.Log(indent + "- - AttachNode's other part allows crossfeed in general.");
-                if (part.NoCrossFeedNodeKey.Length > 0 && attachNode.id.Contains(part.NoCrossFeedNodeKey)) continue;
-
-                if (DEBUG_WALK) Debug.Log(indent + "- - This part allows crossfeed through specifically through this AttachNode.");
-                if (attachNode.attachedPart.NoCrossFeedNodeKey.Length > 0 && attachNode.id.Contains(attachNode.attachedPart.NoCrossFeedNodeKey)) continue;
-
-                if (DEBUG_WALK) Debug.Log(indent + "- -  Part on other side allows flow specifically through this AttachNode.");
-                ret += ProspectForResource(resourceName, attachNode.attachedPart, lines, rDepth + 1, ref visited);
-            }
-
-            // Fuel lines have to be handled specially because they are not in the normal parts tree
-            // and are not connected via AttachNodes:
-            foreach (var fuelLine in lines.Where(fuelLine => part == fuelLine.target && fuelLine.fuelLineOpen && fuelLine.fuelCrossFeed))
-            {
-                if (DEBUG_WALK) Debug.Log(indent + "- Part is target of a fuel line, traversing fuel line upstream.");
-                ret += ProspectForResource(resourceName, fuelLine.parent, lines, rDepth + 1, ref visited);
-            }
-
-            if (DEBUG_WALK) Debug.Log(indent + "Sum from this branch of the recurse tree is " + ret);
-            return ret;
-#pragma warning restore 162
-        }
-
```

Comment:
It took me a lot of work to make the ugly mess above work, but I'm very happy to see it go away.  It was built on a very fragile assumption about SQUAD keeping the same data structure for fuel transference working the same way, which was never going to happen because the way it worked was kind of dumb in some ways.  (Specifically how fuel hoses worked) and therefore we should have expected SQUAD would clean it up later, which they just did.